From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Thu, 23 Jun 2016 01:11:33 -0400
Subject: Actually use distinct NSS directory

several subcommands fail to use the configured NSS directory, and used
the main configuration directory instead.  This should fix these
programs.
---
 programs/look/look.in              | 4 ++--
 programs/pluto/plutomain.c         | 3 ++-
 programs/rsasigkey/rsasigkey.c     | 4 ++--
 programs/showhostkey/showhostkey.c | 2 +-
 4 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/programs/look/look.in b/programs/look/look.in
index 77d7c04..231b839 100755
--- a/programs/look/look.in
+++ b/programs/look/look.in
@@ -99,12 +99,12 @@ echo ROUTING TABLES
 ip -4 route
 ip -6 route
 
-if [ -f @IPSEC_CONFDDIR@/cert9.db ]; then
+if [ -f @IPSEC_NSSDIR@/cert9.db ]; then
     echo NSS_CERTIFICATES
     certutil -L -d sql:@IPSEC_NSSDIR@ | head -4
     certutil -L -d sql:@IPSEC_NSSDIR@ | \
 	egrep -v '^NSS_CERTIFICATES|^$|Trust Attributes|MIME,' | LC_ALL=C sort
-elif [ -f @IPSEC_CONFDDIR@/cert8.db ]; then
+elif [ -f @IPSEC_NSSDIR@/cert8.db ]; then
     echo NSS_CERTIFICATES
     certutil -L -d @IPSEC_NSSDIR@ | head -4
     certutil -L -d @IPSEC_NSSDIR@ | \
diff --git a/programs/pluto/plutomain.c b/programs/pluto/plutomain.c
index f83648d..4e477a4 100644
--- a/programs/pluto/plutomain.c
+++ b/programs/pluto/plutomain.c
@@ -1691,11 +1691,12 @@ void show_setup_plutomain()
 	whack_log(RC_COMMENT, "config setup options:");	/* spacer */
 	whack_log(RC_COMMENT, " ");	/* spacer */
 	whack_log(RC_COMMENT,
-		"configdir=%s, configfile=%s, secrets=%s, ipsecdir=%s, dumpdir=%s, statsbin=%s",
+		"configdir=%s, configfile=%s, secrets=%s, ipsecdir=%s, nssdir=%s, dumpdir=%s, statsbin=%s",
 		oco->confdir,
 		oco->conffile,
 		oco->secretsfile,
 		oco->confddir,
+		oco->nssdb,
 		coredir,
 		pluto_stats_binary == NULL ? "unset" :  pluto_stats_binary);
 
diff --git a/programs/rsasigkey/rsasigkey.c b/programs/rsasigkey/rsasigkey.c
index 2d075ae..365a0ae 100644
--- a/programs/rsasigkey/rsasigkey.c
+++ b/programs/rsasigkey/rsasigkey.c
@@ -201,7 +201,7 @@ int main(int argc, char *argv[])
 			break;
 		case 'c':       /* nss configuration directory */
 		case 'd':       /* -d is used for configdir with nss tools */
-			lsw_conf_confddir(optarg);
+			lsw_conf_nssdb(optarg);
 			break;
 		case 'P':       /* token authentication password */
 			lsw_conf_nsspassword(optarg);
@@ -287,7 +287,7 @@ void rsasigkey(int nbits, int seedbits, const struct lsw_conf_options *oco)
 	realtime_t now = realnow();
 
 	lsw_nss_buf_t err;
-	if (!lsw_nss_setup(oco->confddir, 0, lsw_nss_get_password, err)) {
+	if (!lsw_nss_setup(oco->nssdb, 0, lsw_nss_get_password, err)) {
 		fprintf(stderr, "%s: %s\n", progname, err);
 		exit(1);
 	}
diff --git a/programs/showhostkey/showhostkey.c b/programs/showhostkey/showhostkey.c
index 1341318..de1a9f8 100644
--- a/programs/showhostkey/showhostkey.c
+++ b/programs/showhostkey/showhostkey.c
@@ -453,7 +453,7 @@ int main(int argc, char *argv[])
 	 * processed, and really are "constant".
 	 */
 	const struct lsw_conf_options *oco = lsw_init_options();
-	libreswan_log("using config directory \"%s\"\n", oco->confddir);
+	libreswan_log("using config directory \"%s\"\n", oco->nssdb);
 
 	/*
 	 * Set up for NSS - contains key pairs.
